packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -145,16 +144,6 @@ export function getMiniflareOptions(
 	const workerToDurableObjectClassNamesMap =
 		getWorkerToDurableObjectClassNamesMap(workers);
 
-	// TODO: we only have a single module resolution strategy shared across all workers
-	//       (generated using the first worker's dev environment)
-	//       we should investigate and ideally have potential different resolutions per worker
-	//       see: https://github.com/flarelabs-net/vite-plugin-cloudflare/issues/19
-	const firstWorkerName = workers[0]?.name;
-	invariant(firstWorkerName, 'First worker name not found');
-	const devEnvironment = viteDevServer.environments[firstWorkerName];
-	invariant(devEnvironment, 'First worker dev environment not found');
-	const resolveId = getResolveId(viteConfig, devEnvironment);
-
```

Comment:
This issue went away by itself 🎉 

To tweak the module resolution of a specific worker all you need to do is apply appropriate `overrides` on that worker 🙂